### ISSUE TITLE
More content-length problems with 304 Not Modified response

### ISSUE THREAD

**Issue by henrylaxen** [snapframework/snap-core#242]
```haskell
{-
Dear Snappers,

Looking through the issue tracker for "304" or "content-length" I can
see that there have been ongoing problems.  Well, sorry about that,
but I am about to add to them.  Here is my experience: When I send
back a "304 Not Modified" response, Google chrome is happy and does
not touch the page being displayed.  Firefox, however, is not so
gracious.  It displays a blank page.  I believe this is because
snap-core returns a "Content-Length: 0" header along with the 304
response.  At least that is what Google tells me.  The little program
below demonstrates this.  You should be able to just run this file
through ghc and see what I mean.  In case you are wondering, I am
using the latest (as of Aug 5th, 2015) versions of everything.


ghc-pkg list | grep snap
    digestive-functors-snap-0.6.1.3
    snap-0.14.0.6
    snap-core-0.9.7.2
    snap-loader-static-0.9.0.2
    snap-server-0.9.5.1
    websockets-snap-0.9.2.0

I hope this is helpful.

-}



{-# LANGUAGE OverloadedStrings #-}
module Main where

import           Control.Applicative
import           Control.Monad.IO.Class          ( MonadIO(liftIO) )
import           Snap
import           System.IO

main :: IO ()
main = quickHttpServe $
  path "304" handle304
  <|>
  path "index"  (writeText "index\n")
  <|>
  path "favicon.ico"  (writeText "")

handle304 :: Snap a
handle304 = do
  let
    myResponse = setResponseStatus 304 "Not Modified" emptyResponse
    removedContentLength = deleteHeader "Content-Length" myResponse
  liftIO $ System.IO.hPutStr stderr (show removedContentLength)
  -- finishWith myResponse         -- has Content-Length: 0
  finishWith removedContentLength  -- has Content-Length: 0


{-

/home/henry/haskell> wget -S -q http://localhost:8000/index -O -
  HTTP/1.1 200 OK
  Server: Snap/0.9.5.1
  Date: Wed, 05 Aug 2015 18:56:24 GMT
  Transfer-Encoding: chunked
index
------------------------------------------------------------------------
/home/henry/haskell> wget -S -q http://localhost:8000/304 -O -  
  HTTP/1.1 304 Not Modified
  Content-Length: 0
  Server: Snap/0.9.5.1
  Date: Wed, 05 Aug 2015 18:56:27 GMT
------------------------------------------------------------------------

-}

**Comment by gregorycollins** [comment]
Thanks for the report, could you please confirm the fix works for you?

**Comment by henrylaxen** [comment]
Dear Gregory,

Right now I'm sorry, but I can't.  I would need to upgrade to the latest
versions of snap and snap-server, which cabal is not very enthusiastic about.
I'm sure if it works for you, it will work for me.

Best wishes,
H


Gregory Collins writes:
 > Thanks for the report, could you please confirm the fix works for you?
 > 
 > —
 > Reply to this email directly or view it on GitHub.*
 > 

-- 
Nadine and Henry Laxen   The rest is silence
Villa Alta #6            
Calle Gaviota #10        Never try to teach a pig to sing
Chapala                  It wastes your time  
(376) 765-3181           And it annoys the pig

                             ;\ 
                            |' \ 
         _                  ; : ; 
        / `-.              /: : | 
       |  ,-.`-.          ,': : | 
       \  :  `. `.       ,'-. : | 
        \ ;    ;  `-.__,'    `-.| 
         \ ;   ;  :::  ,::'`:.  `. 
          \ `-. :  `    :.    `.  \ 
           \   \    ,   ;   ,:    (\ 
            \   :., :.    ,'o)): ` `-. 
           ,/,' ;' ,::"'`.`---'   `.  `-._ 
         ,/  :  ; '"      `;'          ,--`. 
        ;/   :; ;             ,:'     (   ,:) 
          ,.,:.    ; ,:.,  ,-._ `.     \""'/ 
          '::'     `:'`  ,'(  \`._____.-'"' 
             ;,   ;  `.  `. `._`-.  \\ 
             ;:.  ;:       `-._`-.\  \`. 
              '`:. :        |' `. `\  ) \ 
                 ` ;:       |    `--\__,' 
                   '`      ,' 
                        ,-'

**Comment by gregorycollins** [comment]
The fix is in snap-core 0.9.8.0. If you were using 0.9.7.2 there should be no reason why you can't upgrade.
